### DISCUSSION TITLE
Setting config at run time

### DISCUSSION THREAD

**Discussion by pavan-bandodekar** [processone/ejabberd#3858] [Questions] [answered]
Hello,

I wanted to know if it is possible to set/change any config variables that we set in ejabberd.yml file, from the code at run time. Basically I want to set websocket_timeout and websocket_ping_interval to some custom values at run time.   Below are some details about tech /stack used.

Library used: Strophe.js
Server side technology: PHP
Client is web browser based

If anyone has any idea about it, please help. Thanks !!

**Comment by licaon-kter** [comment]
You restart ejabberd that often?

Edit .yml via some script and then `ejabberdctl reload_config` to apply the new settings

**Comment by badlop** [comment, marked as answer]
> it is possible to set/change any config variables that we set in ejabberd.yml file, from the code at run time

It is possible using a deprecated function, so this isn't a long-term solution.

```erlang
(ejabberd@localhost)2>  ejabberd_option:websocket_ping_interval().
60000
(ejabberd@localhost)3>  ejabberd_config:set_option(websocket_ping_interval, 12334).
ok
(ejabberd@localhost)4>  ejabberd_option:websocket_ping_interval().                 
12334
```

The preferable solution is to edit ejabberd.yml and reload configuration.
